[DOC] Add GET helm doc (#2185)
grafana/tempo: docs/sources/helm-charts/tempo-distributed/get-started-helm-charts/_index.md

Using a custom namespace solves problems later on because you do not have to overwrite the default namespace.

1. Create a unique Kubernetes namespace, for example `tempo-test`, and switch your local context to use it:

```bash
kubectl create namespace tempo-test
kubens tempo-test
```

> For more details, see the Kubernetes documentation about [Creating a new namespace](https://kubernetes.io/docs/tasks/administer-cluster/namespaces/#creating-a-new-namespace).

Author: Restoring the spacing here. This spacing makes aligns this paragraph with the numbered step. 
```suggestion
  For more details, see the Kubernetes documentation about [Creating a new namespace](https://kubernetes.io/docs/tasks/administer-cluster/namespaces/#creating-a-new-namespace).
```

   helm repo update
   ```

   > **NOTE**: The Helm chart at https://grafana.github.io/helm-charts is a publication of the source code at grafana/tempo.

## Set Helm chart values

Your Helm chart values are set in the `custom.yaml` file. The following example `custom.yaml` file sets the storage and traces options, enables the gateway, and sets the cluster to main. The `traces` configure the distributor's receiver protocols.

> 

Author: ```suggestion

Next, you will:

1. Create a `custom.yaml` file
2. Set your storage values, the example above points to the MinIO instance configured by the chart
3. Set your traces values to configure the receivers on the Tempo distributor
```

    enabled: false
```

### Grafana Enterprise Traces helm chart values

> A complete example for GET is as follows.

Author: ```suggestion
The values in the example below provide configuration values for GET. 
```

```yaml
---
global:
  clusterDomain: 'cluster.local'

multitenancyEnabled: true
enterprise:
  enabled: true
  image:
    tag: weekly-r83-4e9cb510
enterpriseGateway:
  enabled: true
gateway:
  enabled: false
minio:
  enabled: true
storage:
  trace:
    backend: s3
    s3:
      access_key: 'grafana-tempo'
      secret_key: 'supersecret'
      bucket: 'enterprise-traces'
      endpoint: 'tempo-minio:9000'
      insecure: true
  admin:
    backend: s3
    s3:
      access_key_id: 'grafana-tempo'
      secret_access_key: 'supersecret'
      bucket_name: 'enterprise-traces-admin'
      endpoint: 'tempo-minio:9000'
      insecure: true
traces:
  otlp:
    http:
      enabled: true
    grpc:
      enabled: true
distributor:
  config:
    log_received_spans:
      enabled: true

license:
  contents: |
    LICENSEGOESHERE
```

As noted above, please observe the additional `admin` bucket, the `gateway` has been disabled, the `enterpriseGateway` has been enabled, and a license has been specified.

Next, we will:

> 1. Create a `custom.yaml` file

Author: Please delete lines 188-192. I've moved this section earlier in the document.

Reviewer: Okay, we had three places with that wording.  Double check me now, but it looks okay to me.

    enabled: false
```

### Grafana Enterprise Traces helm chart values

The values in the example below provide configuration values for GET.
These values include an additional `admin` bucket, the `gateway` has been disabled, the `enterpriseGateway` has been enabled, and a license has been specified.

```yaml
---
global:
  clusterDomain: 'cluster.local'

multitenancyEnabled: true
enterprise:
  enabled: true
  image:
    tag: v2.0.1
enterpriseGateway:
  enabled: true
gateway:
  enabled: false
minio:
  enabled: true
storage:
  trace:
    backend: s3
    s3:
      access_key: 'grafana-tempo'
      secret_key: 'supersecret'
      bucket: 'enterprise-traces'
      endpoint: 'tempo-minio:9000'
      insecure: true
  admin:
    backend: s3
    s3:
      access_key_id: 'grafana-tempo'
      secret_access_key: 'supersecret'
      bucket_name: 'enterprise-traces-admin'
      endpoint: 'tempo-minio:9000'
      insecure: true
traces:
  otlp:
    http:
      enabled: true
    grpc:
      enabled: true
distributor:
  config:
    log_received_spans:
      enabled: true

license:
  contents: |
    LICENSEGOESHERE
```

#### Enterprise license configuration

> If you are GET, you need to configure a license, by adding the license to the `custom.yaml` file or by using a secret that contains the license.

Author: ```suggestion
If you are GET, you need to configure a license, by adding the license to the `custom.yaml` file or by using a secret that contains the license.
Only one of these options should be used. 
```

> **NOTE**: The [Set up GET instructions](https://grafana.com/docs/enterprise-traces/latest/setup/#obtain-a-get-license) explain how to obtain a license.

First, you can specify the license text in the `custom.yaml` values file created above, in the `license:` section.

```yaml
license:
  contents: |
    LICENSEGOESHERE
```

> Secondly, you may use a secret that contains the license content that is referenced.

Author: ```suggestion
If you do not with to specific the license in the `custom.yaml` file, you can use a secret that contains the license content that is referenced.
```

    enabled: false
```

### Grafana Enterprise Traces helm chart values

The values in the example below provide configuration values for GET.

```yaml
---
global:
  clusterDomain: 'cluster.local'

multitenancyEnabled: true
enterprise:
  enabled: true
  image:

>     tag: weekly-r83-4e9cb510

Author: @zalegrala Does this value need to be updated?

enterpriseGateway:
  enabled: true
gateway:
  enabled: false
minio:
  enabled: true
storage:
  trace:
    backend: s3
    s3:
      access_key: 'grafana-tempo'
      secret_key: 'supersecret'
      bucket: 'enterprise-traces'
      endpoint: 'tempo-minio:9000'
      insecure: true
  admin:
    backend: s3
    s3:
      access_key_id: 'grafana-tempo'
      secret_access_key: 'supersecret'
      bucket_name: 'enterprise-traces-admin'
      endpoint: 'tempo-minio:9000'
      insecure: true
traces:
  otlp:
    http:
      enabled: true
    grpc:
      enabled: true
distributor:
  config:
    log_received_spans:
      enabled: true

> license:

Reviewer 2: It looks like this snippet that embeds the license in values.yaml doesn't match the recommendations below which are a separate file or a k8s secret.  Should it be removed?

Reviewer: Oh do you mean I should reorder the options below and preference the `external` secret option?

license:
  contents: |
    LICENSEGOESHERE
```

#### Enterprise license configuration

> If you are GET, you need to configure a license, by adding the license to the `custom.yaml` file or by using a secret that contains the license.

Author: ```suggestion
If you are using GET, you need to configure a license, by adding the license to the `custom.yaml` file or by using a secret that contains the license.
```

1. Copy the example above and paste it into the custom.yaml file.
1. Save the file.

#### Enterprise license configuration

> If you are using Grafana Enterprise Traces, you will need to configure a license, which may be done in one of two ways.

Author: ```suggestion
If you are using Grafana Enterprise Traces, you need to configure a license, which may be done in one of two ways.

>**NOTE**: The [Set up GET instructions](https://grafana.com/docs/enterprise-traces/latest/setup/#obtain-a-get-license) explain how to obtain a license. 

```

First, you may specify the license text in the `custom.yaml` values file.

```yaml
license:
  contents: |
    LICENSEGOESHERE
```

Secondly, you may use a secret that contains the license content.

1. Create the secret

```bash
kubectl create secret generic tempo-license --from-file=license.jwt

> ```

Author: This code block needs to be indented to it appears inline with the number. Otherwise, the numbering will be reset instead of being continuous. 

Same with the number on line 220.

Reviewer: Okay, should be intended now, hows that?

    enabled: false
```

### Optional: Add custom configurations

You can use a YAML file, like `custom.yaml`, to store custom configuration options that override the defaults present in the Helm chart.
The [tempo-distributed Helm chart's README](https://github.com/grafana/helm-charts/blob/main/charts/tempo-distributed/README.md) contains a list of available options.

> Any values that are enabled in `values.yaml` are on by default.

Author: I'm not happy with this wording. I added the sentence because I wanted users to understand that the values.yaml file provides the defaults for the helm installation. Oh. I should just state that. 
```suggestion
The `values.yaml` files provides the defaults for the helm chart. 
```

tempo-minio-6c4b66cb77-wjfpf                1/1     Running     0             86m
tempo-querier-6cb474546-cwlkz               1/1     Running     0             86m
tempo-query-frontend-6d6566cbf7-pcwg6       1/1     Running     0             86m
tempo-tokengen-job-58jhs                    0/1     Completed   0             86m
```

Note that the `tempo-tokengen-job` has emitted a log message containing the initial admin token.

Retrieve the token with this handy one-liner.

```
kubectl get pods | awk '/.*-tokengen-job-.*/ {print $1}' | xargs -I {} kubectl logs {} | awk '/Token:\s+/ {print $2}'
```

Or simply get the logs for the `tokengen` pod, for example.

```
kubectl logs tempo-tokengen-job-58jhs
```

## Next step

The next step is to test your Tempo installation by sending trace data to Grafana. You can use the [Set up a test application for a Tempo cluster](https://grafana.com/docs/tempo/latest/setup/set-up-test-app/) document for step-by-step instructions.

If you already have Grafana available, you can add a Tempo data source using the URL fitting to your environment. For example:
`http://tempo-query-frontend.trace-test.svc.cluster.local:3100`

> Enterprise users may wish to install the Enterprise Traces plugin in their Enterprise Grafana instance to allow configuration of tenants, tokens, and access policies. Once a user, and access policy have been created using the plugin, a datasource can be configured to point at `http://tempo-enterprise-gateway.tempo-test.svc.cluster.local:3100`.

Author: ```suggestion
Enterprise users may wish to [install the Enterprise Traces plugin](https://grafana.com/docs/enterprise-traces/latest/setup/setup-get-plugin-grafana/) in their Grafana Enterprise instance to allow configuration of tenants, tokens, and access policies. Once a user, and access policy have been created using the plugin, a datasource can be configured to point at `http://tempo-enterprise-gateway.tempo-test.svc.cluster.local:3100`.
```
